includes/class-sensei-course-structure.php

```diff
@@ -117,18 +127,22 @@ private function prepare_lesson( WP_Post $lesson_post ) {
 			'type'  => 'lesson',
 			'id'    => $lesson_post->ID,
 			'title' => $lesson_post->post_title,
+			'draft' => 'draft' === $lesson_post->post_status,
 		];
 	}
 
 	/**
 	 * Get the lessons for a module.
 	 *
-	 * @param int $module_term_id Term ID for the module.
+	 * @param int  $module_term_id         Term ID for the module.
+	 * @param bool $published_lessons_only Only include published lessons.
 	 *
 	 * @return WP_Post[]
 	 */
-	private function get_module_lessons( int $module_term_id ) {
-		$lessons_query = Sensei()->modules->get_lessons_query( $this->course_id, $module_term_id, 'any' );
+	private function get_module_lessons( int $module_term_id, bool $published_lessons_only ) {
+		$post_status = $published_lessons_only ? self::PUBLISHED_POST_STATUSES : 'any';
```

Comment:
We can pass `$lesson_post_status` through these methods instead of the `$published_lessons_only` flag

Reply:
✅ 6409a67d8569901648c9947352c9ad9b29415d29